client/app/configuration/deployment-maps/DeploymentMapTargetController.js

```diff
@@ -267,20 +247,39 @@ angular.module('EnvironmentManager.configuration').controller('DeploymentMapTarg
       return _.map(deploymentMap.Value.DeploymentTarget, 'ServerRoleName');
     }
 
-    function validateDataForSave() {
-      var valid = { result: true };
+    function validateDataForPuppetRole() {
+      if (vm.target.ASG.LaunchConfig.AMIPlatform == 'Linux' && typeof vm.target.ASG.LaunchConfig.PuppetRole == 'undefined') {
```

Comment:
The best thing here, would be if we had drop downs for available roles... but that is totally out of scope and I appreciate that. 

Do you think it's worth changing this to `!!vm.target.ASG.LaunchConfig.PuppetRole.trim()`?